rpc-protocol/src/main/java/org/sonarsource/sonarlint/core/rpc/protocol/backend/initialize/FeatureFlagsDto.java

```diff
@@ -19,87 +19,76 @@
+import java.util.Set;
+
 /**
+ * @deprecated use InitializeParams constructor that accepts Set<BackendCapability> directly
  * Optional features toggles of the backend. To ease transition or to accommodate different client needs.
  */
+@Deprecated(since = "10.19", forRemoval = true)
 public class FeatureFlagsDto {
 
-  private final boolean shouldManageSmartNotifications;
-  @Deprecated
-  // not controllable anymore, it is the backend's responsibility to decide
-  private final boolean taintVulnerabilitiesEnabled;
-  private final boolean shouldSynchronizeProjects;
-  private final boolean shouldManageLocalServer;
-  private final boolean enableSecurityHotspots;
-  private final boolean shouldManageServerSentEvents;
-  private final boolean enableDataflowBugDetection;
-  private final boolean shouldManageFullSynchronization;
-  private final boolean enableTelemetry;
-  private final boolean canOpenFixSuggestion;
-  private final boolean enableMonitoring;
+  private final Set<BackendCapability> backendCapabilities = EnumSet.noneOf(BackendCapability.class);
 
+  /**
+   * @deprecated call new constructor accepting enums
+   *
+   */
+  @Deprecated(since = "10.19")
   public FeatureFlagsDto(boolean shouldManageSmartNotifications, boolean taintVulnerabilitiesEnabled, boolean shouldSynchronizeProjects, boolean shouldManageLocalServer,
     boolean enableSecurityHotspots, boolean shouldManageServerSentEvents, boolean enableDataflowBugDetection, boolean shouldManageFullSynchronization, boolean enableTelemetry,
     boolean canOpenFixSuggestion, boolean enableMonitoring) {
-    this.shouldManageSmartNotifications = shouldManageSmartNotifications;
-    this.taintVulnerabilitiesEnabled = taintVulnerabilitiesEnabled;
-    this.shouldSynchronizeProjects = shouldSynchronizeProjects;
-    this.shouldManageLocalServer = shouldManageLocalServer;
-    this.enableSecurityHotspots = enableSecurityHotspots;
-    this.shouldManageServerSentEvents = shouldManageServerSentEvents;
-    this.enableDataflowBugDetection = enableDataflowBugDetection;
-    this.shouldManageFullSynchronization = shouldManageFullSynchronization;
-    this.enableTelemetry = enableTelemetry;
-    this.canOpenFixSuggestion = canOpenFixSuggestion;
-    this.enableMonitoring = enableMonitoring;
+    addIfTrue(shouldManageSmartNotifications, BackendCapability.SMART_NOTIFICATIONS);
+    addIfTrue(shouldSynchronizeProjects, BackendCapability.PROJECT_SYNCHRONIZATION);
+    addIfTrue(shouldManageLocalServer, BackendCapability.EMBEDDED_SERVER);
+    addIfTrue(enableSecurityHotspots, BackendCapability.SECURITY_HOTSPOTS);
+    addIfTrue(shouldManageServerSentEvents, BackendCapability.SERVER_SENT_EVENTS);
+    addIfTrue(enableDataflowBugDetection, BackendCapability.DATAFLOW_BUG_DETECTION);
+    addIfTrue(shouldManageFullSynchronization, BackendCapability.FULL_SYNCHRONIZATION);
+    addIfTrue(enableTelemetry, BackendCapability.TELEMETRY);
+    addIfTrue(enableMonitoring, BackendCapability.MONITORING);
```

Comment:
This code looks very similar to what was added in `InitializeParams`. But no big deal as this class should soon be removed

Reply:
Yes, I don't see an easy way to remove this similar code. Let's just remove it later with the class later.